packages/theme-compiler/src/helpers.js

```diff
@@ -7,8 +7,13 @@ const autoprefixer = require('postcss')().use(require('autoprefixer'));
 const clean = new (require('clean-css'))({ returnPromise: true, level: '2' });
 const path = require('path');
 
-const wrap = (name, style) => `elf.${name.indexOf('-') > 0 ? 'custom'
-: 'native'}Styles.define('${name}', '${style.replace(/'/g, '\\\'')}');\n`;
+const wrap = (name, style, isEvent) => {
+  if(isEvent) {
+    return `dispatchEvent(new CustomEvent('ef.${name.indexOf('-') > 0 ? 'custom'
+    : 'native'}Styles.define', { detail: { name: '${name}', styles: '${style.replace(/'/g, '\\\'')}' }}));\n`;
+  }
+  return `elf.${name.indexOf('-') > 0 ? 'custom' : 'native'}Styles.define('${name}', '${style.replace(/'/g, '\\\'')}');\n`;
```

Comment:
The condition for checking the custom/native styles should be moved out of the template literal

Reply:
Updated.

Reply:
@Theeraphat-Sorasetsakul, You should use the `eventName` on line 19

Reply:
Updated.